funnel/models/event.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+from . import db, BaseMixin
+from .space import ProposalSpace
+from .user import User
+import random
+import uuid
+from sqlalchemy.ext.associationproxy import association_proxy
+
+__all__ = ['Event', 'TicketType', 'Participant', 'Attendee', 'SyncTicket']
+
+PRINTABLE_ASCII = map(chr, range(32, 127))
+ATTENDEE_TABLE_NAME = 'attendee'
+EVENT_TICKET_TYPE_TABLE_NAME = 'event_ticket_type'
+
+
+def make_key():
+    key = str(uuid.uuid4()).replace('-', '')
+    return ''.join(random.sample(key, len(key)))
+
+
+def make_public_key():
+    return make_key()[:8]
+
+
+def make_private_key():
+    return make_key()[:8]
+
+
+event_ticket_type = db.Table(EVENT_TICKET_TYPE_TABLE_NAME, db.Model.metadata,
+    db.Column('event_id', db.Integer, db.ForeignKey('event.id')),
+    db.Column('ticket_type_id', db.Integer, db.ForeignKey('ticket_type.id'))
+)
+
+
+class Event(BaseMixin, db.Model):
+    """ A discrete event under a proposal space.
+        For instance, a space could be associated with a workshop and a two-day conference.
+        The workshop constitutes as one event and each day of the conference
+        constitutes as an independent event.
+        This is done to allow distinguishing participants based on
+        on the tickets they have, given a participant may have a ticket
+        for only the workshop or a single day of the conference.
+        An event is associated with multiple ticket types,
+        which helps make the distinction between participants
+    """
+    __tablename__ = 'event'
+
+    name = db.Column(db.Unicode(80), nullable=False)
+    proposal_space_id = db.Column(None, db.ForeignKey('proposal_space.id'), nullable=False)
+    proposal_space = db.relationship(ProposalSpace,
+        backref=db.backref('events', cascade='all, delete-orphan', lazy='dynamic'))
+    ticket_types = db.relationship("TicketType", secondary=EVENT_TICKET_TYPE_TABLE_NAME, lazy='dynamic')
```

Comment:
`secondary` should preferably refer to the table object itself, not it's name (SQLAlchemy tries behind the scenes to map a string to the table object).